sentry_sdk/integrations/falcon.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, Callable, Dict
+
+import falcon
+import falcon.api_helpers
+import sentry_sdk.integrations
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations._wsgi_common import RequestExtractor
+from sentry_sdk.integrations.wsgi import SentryWsgiMiddleware
+from sentry_sdk.utils import capture_internal_exceptions, event_from_exception
+
+
+class FalconRequestExtractor(RequestExtractor):
+
+    def env(self):
+        return self.request.env
+
+    def cookies(self):
+        return self.request.cookies
+
+    def raw_data(self):
+        # As request data can only read once we won't make this available
```

Comment:
I think it's ok if you return a placeholder string to just tell the user that "raw data" was part of the request.

In the end we will not send the raw payload to sentry anyway as we can't strip sensitive data from arbitrary blobs (we also don't send *contents* of file uploads for that reason).